Homework-№4-from-28.09.2022/task3.c

```diff
@@ -0,0 +1,205 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+
+#define maxSizeOfEntry 101
+
+typedef struct entry {
+    char phoneNumber[maxSizeOfEntry];
+    char name[maxSizeOfEntry];
+} dataBase;
+
+void instructionPrint() {
+    printf("This program has these commands:\n");
+    printf("0 - exit\n");
+    printf("1 - add an entry (name and phone number)\n");
+    printf("2 - print all available records\n");
+    printf("3 - find phone by name\n");
+    printf("4 - find name by phone\n");
+    printf("5 - save the current data to a file\n");
+}
+
+int readCommand() {
+    printf("Please, enter the command:");
+    int command = 0;
+    scanf("%d", &command);
+    while (!((command < 6) && (command > -1))) {
+        printf("Wrong format! Please, try enter\n");
+        printf("the program number again:");
+        scanf("%d", &command);
+    }
+    return command;
+}
+
+int readDataFromFile(dataBase **entries, char *nameOfFile, int *cntOfEntries) {
+    FILE *file = fopen(nameOfFile, "r");
+    if (file == NULL) {
+        return -2;
+    }
+    char checkString[] = "phoneNumbers names";
+    char compareFirstString[19] = {0};
+    for (int i = 0; i < 18; ++i) {
+        compareFirstString[i] = fgetc(file);
+    }
+
+    if (strcmp(compareFirstString, checkString)) {
+        printf("Incorrect file!\n");
+        return -3;
+    }
+    fclose(file);
+    file = fopen(nameOfFile, "r");
+    for (int i = 0; i < 2; ++i) {
+        fscanf(file, "%s", checkString);
+    }
+    while (!feof(file)) {
+        fscanf(file, "%s", entries[*cntOfEntries]->phoneNumber);
+        fscanf(file, "%s", entries[*cntOfEntries]->name);
+        ++*cntOfEntries;
+    }
+    fclose(file);
+    return 0;
+}
+
+void command1(dataBase **entries, int *cntOfEntries) {
```

Comment:
```suggestion
void addAnEntry(dataBase **entries, int *cntOfEntries) {
```
Название должно соответствовать тому, что функция делает. Измени имена остальных функций `command`.